root/static/scripts/register.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * @flow
+ * Copyright (C) 2021 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import $ from 'jquery';
+
+import debounce from './common/utility/debounce';
+
+$(function () {
+  function warnAboutEmailAsUsername() {
+    const username =
+    $('#id-register\\\.username').val();
+    const isPossibleEmail = /\w+@\w+\.\w+/.test(username);
```

Comment:
This uses an intentionally wide regexp because, well, it doesn't really cause any problems if the string happens to be an invalid username - the user can just ignore the warning and move on.